src/datamodel_code_generator/config.py

```
@@ -201,9 +202,36 @@ class ParserConfig(BaseModel):
         class Config:
             """Pydantic v1 model config."""
 
-            extra = "forbid"
+            extra = "allow"
             arbitrary_types_allowed = True
 
+    @classmethod
+    def from_options(cls, options: dict[str, Any]) -> Self:
+        """Create a ParserConfig from options dict with version-compatible validation."""
+        if is_pydantic_v2():
+            from datamodel_code_generator.model.base import (  # noqa: PLC0415
```

Ruff (0.14.10)

Unused `noqa` directive (non-enabled: `PLC0415`)

Remove unused `noqa` directive

(RUF100)